pkg/scheduler/factory/factory.go

```diff
@@ -308,19 +309,26 @@ func NewConfigFactory(
 		podQueue:   c.podQueue,
 	}
 
-	ch := make(chan os.Signal, 1)
-	signal.Notify(ch, compareSignal)
+	if resyncIntervalInSec != 0 {
+		ch := make(chan os.Signal, 1)
+		signal.Notify(ch, compareSignal)
 
-	go func() {
-		for {
-			select {
-			case <-c.StopEverything:
-				return
-			case <-ch:
-				comparer.Compare()
+		ticker := time.NewTicker(time.Duration(resyncIntervalInSec) * time.Second)
+		defer ticker.Stop()
```

Comment:
Doesn't the ticker stop shortly after this point when the function returns? Should this be in the `go func()`?

Reply:
good catch!